virtualizarr/xarray.py

```diff
@@ -135,6 +143,42 @@ def open_virtual_dataset(
             filetype=filetype,
             reader_options=reader_options,
         )
+
+        if filetype.name.lower() == "hdf5" or filetype.name.lower() == "netcdf4":
+            hdf_groups = [
+                k.removesuffix(".zgroup")
+                for k in vds_refs["refs"].keys()
+                if ".zgroup" in k
+            ]
+            if len(hdf_groups) > 1:
+                if group is None:
+                    raise ValueError(
+                        f"Multiple HDF Groups found. Must specify group= keyword to select one of {hdf_groups}"
+                    )
+                else:
+                    # Ensure formatting consistent with kerchunk keys
+                    if not group.endswith("/"):
+                        group += "/"
+                    if group.startswith("/"):
+                        group = group.removeprefix("/")
+
+                if group not in hdf_groups:
+                    raise ValueError(
+                        f"Group {group} not found. Available groups are: {hdf_groups}"
+                    )
+
+                # Filter by group prefix and remove prefix from all keys
+                groupdict = {
+                    k.removeprefix(group): v
+                    for k, v in vds_refs["refs"].items()
+                    if k.startswith(group)
+                }
```

Comment:
I was a bit confused as to what to add, so I just put some code comments that I think help clarify `KerchunkStoreRefs` and `KerchunkArrRefs`. I admit I'm really new to type checking. I guess it could be nice to do some sort of schema validation for the correct keys in these dicts...

Reply:
> I was a bit confused as to what to add, so I just put some code comments that I think help clarify `KerchunkStoreRefs` and `KerchunkArrRefs`.

Okay that's fine for now!

> I admit I'm really new to type checking.

Type checking is great. In fact I think most of the value comes from how much clearer it makes the intention of the code, even if you never actually ran the static type checker!

This is a weird example of type checking though, it is actually more of a schema validation problem as you say. And yes kerchunk should be doing schema validation on all its internal types, but it doesn't, so we have to deal with the resultant bugs.